runtime/doc/develop.txt

```diff
@@ -233,4 +225,23 @@ _not_ a Buffer). The common {action} "list" indicates that it lists all
 bufs (plural) in the global context.
 
 
+EXTERNAL UI 						*dev-ui*
+
+External UIs should be aware of the |api-contract|. In particular, future
+versions of Nvim may add optional, new items to existing events. The API is
+strongly backwards-compatible, but clients must not break if new fields are
+added to existing events.
+
+External UIs are expected to implement some common features.
+
+- Users may want to configure UI-specific options. The UI should publish the
+  |GUIEnter| autocmd after attaching to Nvim: >
+    doautocmd GUIEnter
```

Comment:
Similarily the remote ui infrastructure should fire this event, rather than every single ui client remembering to do this.